docs/source/intermediate/benchmarking.rst

```diff
@@ -0,0 +1,285 @@
+============
+In Thunder there are a two of ways of benchmarking the compiler:
+
+- One is by running a synthetic end-to-end training of a model from `LitGPT <https://github.com/Lightning-AI/litgpt/>`__ ; and
+- the second is by microbenchmarking specific snippets of code.
+
+Before starting, you need to install Thunder and the devel packages with::
+
+  pip install -r requirements/devel.txt
+  pip install -e .
+
+LitGPT benchmarks
+=================
+
+The esiest way to run a single benchmark in Thunder is by running an instance of the LitGPT end-to-end training script that can be found in ``thunder/benchmarks/benchmark_litgpt.py``.
+
+To run a benchmark all we need is the following command::
+
+  python thunder/benchmarks/benchmark_litgpt.py --model_name <model name> --compile "thunder"
+
+
+All the command line options can be queried by passing ``--help`` as argument or `can be seen here <https://github.com/Lightning-AI/lightning-thunder/blob/main/thunder/benchmarks/benchmark_litgpt.py#L103-L120>`_. However, the most important options for single GPU benchmarks are:
+
+- ``--compile``: specifies the compile mode for the run
+- ``--model_name``: LitGPT model name, a list of these can be found `here <https://github.com/Lightning-AI/litgpt/?tab=readme-ov-file#choose-from-20-llms>`_
+- ``--n_layers``: specifies the number of layers in the model and can be useful to test a reduced version of the model in the case the original version does not fit into memory.
+- ``--nsys_enabled``: inserts markers to profile the run with `NVIDIA Nsight System <https://developer.nvidia.com/nsight-systems/get-started>`_
+
+The output from this end-to-end benchmark will look something like this::
+
+  Time to instantiate model: 0.12 seconds.
+  iter 0: loss 10.5625, iter time: 62938.98ms, t: 4096
+  ...
+  iter 44: loss 4.6562, iter time: 829.17ms, t: 4096
+  Model name: Llama-2-7b-hf
+  Seq Length: 4096
+  Micro BS: 1
+  Global BS: 1
+  Number of Layers: 32
+  Number of parameters: 3.50B
+  Distributed Mode: none
+  Compiler: thunder
+  Average iter time: 811.19 ms
+  Memory used: 33.42 GB
+  Tokens/s: 5046.46
+  Tokens/s/GPU: 5046.46
+  TFLOP/s: 118.27
+
+Compile options
+---------------
+
+With the ``--compile`` option, you can test:
+
+- torch.compile by specifying ``inductor``
+- torch eager mode by specifying ``eager``, or
+- Thunder by specifying ``thunder``
+
+To customize Thunder executors, in addition to nvFuser, you can append the any combination of the following to the string with an underscore:
+
+- ``inductor_cat``
+- ``cudnn``
+- ``transformerengine``
+
+As an example, if you want to use cudnn as executor your terminal will looks something like::
+
+  python thunder/benchmarks/benchmark_litgpt.py --model_name Llama-2-7b-hf  --compile thunder_cudnn
+
+
+and if you are testing ``torch.compile`` then it will look something like this::
+
+  python thunder/benchmarks/benchmark_litgpt.py --model_name Llama-2-7b-hf  --compile inductor
+
+PyTest benchmarks
```

Comment:
Is this the official spelling of pytest?

Reply:
No it's not, I took some liberty but you're right it's better to keep to the official spelling